### PR TITLE
feat(federatedfilesharing): log errors, as suggested by @maxbes

### PR DESCRIPTION
**Pull request by faust64**
Adds logs to the federatedfilesharing app, which should help in figuring out why a requests might have failed.

### CODE REVIEW

File: apps/federatedfilesharing/lib/Notifications.php [hunk 1]
```diff
@@ -123,7 +123,17 @@ public function sendRemoteShare($token, $shareWith, $name, $remoteId, $owner, $o
 				$event = new FederatedShareAddedEvent($remote);
 				$this->eventDispatcher->dispatchTyped($event);
 				return true;
+			} else {
+				\OC::$server->getLogger()->info(
```
Comment (rullzer) [inline apps/federatedfilesharing/lib/Notifications.php:127]: Please inject the logger in the constructor. So you don't have to do all these \OC::$server calls.

**Review by rullzer — CHANGES_REQUESTED** [review]

File: apps/federatedfilesharing/lib/Notifications.php (hunk 1 above)
Comment (faust64) [reply apps/federatedfilesharing/lib/Notifications.php]: Sure. I've been looking at how it was done in `apps/user_ldap/lib/User_LDAP.php`, which also has those \OC::$server.  `apps/user_ldap/lib/Group_LDAP.php` does init its logger in constructor, though.
Not sure including `OCP\ILogger` is required. In doubt, I'll stick to what works in other apps.
Let me know if there's other stuff I could clean up.

File: apps/federatedfilesharing/lib/Notifications.php [hunk 2]
```diff
@@ -73,6 +78,7 @@ public function __construct(
 		$this->httpClientService = $httpClientService;
 		$this->discoveryService = $discoveryService;
 		$this->jobList = $jobList;
+		$this->logger = OC::$server->getLogger();
```
Comment (rullzer) [inline apps/federatedfilesharing/lib/Notifications.php:81]: Ah sorry I should have been more clear.
Have the contstructor take the ILogger as argument. Then it gets injected (with automatically setting the app etc as well).

Comment (faust64) [reply apps/federatedfilesharing/lib/Notifications.php]: oh, allright. Then something like `apps/weather_status/lib/Service/WeatherStatusService.php`?

**Comment by rullzer** [comment]
Cool stuff. it seems the test need a fix as well. Could you have a look at that?

**Comment by faust64** [comment]
There's some ratelimit issue apparently, and a timeout during some mariadb initialization that I can find in other jobs as well, ... I guess the previous commit should be enough. Otherwise let me know @rullzer .

**Comment by faust64** [comment]
anyone?

**Review by rullzer — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
👍

**Pull request merged**